### ISSUE TITLE
Use languages default values when configuring language instead of global defaults

### ISSUE THREAD

**Issue by akinsho** [b3nj5m1n/kommentary#31]
First off thanks for this plugin 👍🏿 it's been a great and generally bug free addition for me.

```lua
    use {
      "b3nj5m1n/kommentary",
      config = function()
        require("kommentary.config").configure_language("lua", {prefer_single_line_comments = true})
      end
    }

```
In order to get round the issue in #3 I added the above configuration but only for lua. However with this config `//` is used instead as the comment character.

Note that when I use `default` as the language this works correctly, but ideally since this is a lua specific issue I'd rather not change the behaviour for all files.
```lua
 require("kommentary.config").configure_language("default", {prefer_single_line_comments = true})
```

**Comment by b3nj5m1n** [comment]
Hi, this is actually the expected behaviour, since calling configure_language will use the default values and only overwrite the fields you explicitly specify, default values means the global default in this case, not the language specific default.

So when you call
```lua
require("kommentary.config").configure_language("lua", {prefer_single_line_comments = true})
```

It will overwrite all other values, the important one being the single line comment string, with the global defaults, which would be // for the single line comment string. To solve this, you just have to specify the single line comment string as well:
```lua
require("kommentary.config").configure_language("lua", {single_line_comment_string = "\"", prefer_single_line_comments = true})
```

We can probably agree that this is less than ideal, really we'd want the configure language function to fill in defaults from the language where possible, I just haven't gotten around to implementing that. We'll see when I find time for it, for now I'm afraid you're stuck with the above solution.

**Comment by akinsho** [comment]
@b3nj5m1n thanks for the explanation and the work around makes sense. Seems like one for `vim.tbl_deep_extend` when using the config function.

**Comment by b3nj5m1n** [comment]
This should be fixed with bcdc1222075aefe7a77aabe5b7315e71be5e5560, I briefly tested it, and it seems to work fine, if you still have any problems, please let me know and reopen the issue.

Also, apologies for taking ages for something as simple as this.

**Comment by akinsho** [comment]
@b3nj5m1n thanks for looking at this 🙌🏿 .
I just tried this out, it initially error then I updated again and think you fixed it and now it works fine 👍🏿
